src/Features/Core/Portable/StackTraceExplorer/ParsedStackFrame.cs

```diff
@@ -0,0 +1,132 @@
+{
+    /// <summary>
+    /// A line of text that was parsed by <see cref="StackTraceAnalyzer" />
+    /// to provide metadata bout the line. 
+    /// </summary>
+    internal class ParsedStackFrame : ParsedFrame
+    {
+        public ParsedStackFrame(string originalText, TextSpan classSpan, TextSpan methodSpan, TextSpan argsSpan)
+            : base(originalText)
+        {
+            Contract.ThrowIfTrue(classSpan.IsEmpty);
+            Contract.ThrowIfTrue(methodSpan.IsEmpty);
+
+            ClassSpan = classSpan;
+            MethodSpan = methodSpan;
+            ArgsSpan = argsSpan;
+        }
+
+        /// <summary>
+        /// The full classname parsed from the line. 
+        /// e.x: [|Microsoft.CodeAnalysis.Editor.CallstackExplorer|].Example(arg1, arg2)
+        /// </summary>
+        public TextSpan ClassSpan { get; }
+
+        /// <summary>
+        /// The method name span
+        /// e.x: Microsoft.CodeAnalysis.Editor.CallstackExplorer.[|Example|](arg1, arg2)
+        /// </summary>
+        public TextSpan MethodSpan { get; }
+
+        /// <summary>
+        /// The span of comma seperated arguments.
+        /// e.x: Microsoft.CodeAnalysis.Editor.CallstackExplorer.Example([|arg1, arg2|])
+        /// </summary>
+        public TextSpan ArgsSpan { get; }
+
+        public virtual async Task<ISymbol?> ResolveSymbolAsync(Solution solution, CancellationToken cancellationToken)
+        {
+            var className = GetClassText(includeTrailingDot: false);
+            var methodName = GetMethodText();
+
+            foreach (var project in solution.Projects)
+            {
+                var service = project.GetLanguageService<IStackTraceExplorerService>();
+                if (service is null)
+                {
+                    continue;
+                }
+
+                var metadataName = service.GetClassMetadataName(className);
+                var memberName = service.GetMethodSymbolName(methodName);
+
+                var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
+                var type = compilation.GetTypeByMetadataName(metadataName);
+                if (type is null)
+                {
+                    continue;
+                }
+
+                var members = type.GetMembers();
+                var matchingMembers = members.WhereAsArray(m => MemberMatchesMethodName(m, memberName));
+                if (matchingMembers.Length == 0)
+                {
+                    continue;
+                }
+
+                if (matchingMembers.Length == 1)
+                {
+                    return matchingMembers[0];
+                }
+            }
+
+            return null;
+
+            static bool MemberMatchesMethodName(ISymbol member, string memberToSearchFor)
+            {
+                var displayName = member.ToDisplayString();
+                var dotIndex = displayName.LastIndexOf(".");
```

Comment:
this doesn't seem viable.  the LastIndexOf "." could be inside something like a parameter type.